File: from-0701-to-0800/0713-subarray-product-less-than-k/solution.go
func numSubarrayProductLessThanK(nums []int, k int) int {
    if k < 0 {
        return 0
    }
    result := 0
    product := 1
    left, right := 0, 0

    for right < len(nums) {
        product *= nums[right]

        for product >= k && left <= right {
            product /= nums[left]
            left += 1
        }

        result += (right - left + 1)
        right += 1
    }
    return result
}

File: from-0701-to-0800/0713-subarray-product-less-than-k/solution.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k < 1:
            return 0

        result = 0
        product = 1
        left, right = 0, 0

        while right < len(nums):
            product *= nums[right]
            
            while product >= k and left <= right:
                product //= nums[left]
                left += 1

            result += (right - left + 1)
            right += 1

        return result